src/v/cloud_storage/cache_service.cc

```diff
@@ -37,6 +37,53 @@ cache::cache(
 
 uint64_t cache::get_total_cleaned() { return _total_cleaned; }
 
+bool cache::is_tmp_file(const ss::sstring& filename) {
+    static const ss::sstring tmp_suffix = ".part";
+    auto filename_suffix = filename.substr(
+      filename.length() - tmp_suffix.length(), filename.length());
```

Comment:
Needs some protection against filenames shorter than tmp_suffix. Also I think this can be done without allocating a new string (e.g. by creating a string_view from filename or accepting it as a string_view).

Reply:
Done!